sim/mage/fire/pyroblast.go

```diff
@@ -0,0 +1,146 @@
+			baseDamage := 1.5 * Mage.ScalingBaseDamage
+			result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
+			spell.WaitTravelTime(sim, func(sim *core.Simulation) {
+				if result.Landed() {
+					Mage.PyroblastDot.Cast(sim, target)
+					spell.DealDamage(sim, result)
+					//pyroblastDot.SpellMetrics[target.UnitIndex].Hits++
+					//pyroblastDot.SpellMetrics[target.UnitIndex].Casts = 0
+					/* The 2 above metric changes should show how many ticks land
+					without affecting the overall pyroblast cast metric
+					*/
+				}
+			})
+		},
+	})
+
+	Mage.PyroblastDot = Mage.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 11366}.WithTag(1),
+		SpellSchool:    core.SpellSchoolFire,
+		ProcMask:       core.ProcMaskSpellDamage,
+		Flags:          mage.SpellFlagMage,
+		ClassSpellMask: mage.MageSpellPyroblastDot,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				NonEmpty: true,
+			},
+		},
+
+		DamageMultiplier: 1,
+		CritMultiplier:   Mage.DefaultSpellCritMultiplier(),
+		ThreatMultiplier: 1,
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "PyroblastDoT",
+			},
+			NumberOfTicks:       4,
+			TickLength:          time.Second * 3,
+			BonusCoefficient:    0.180,
+			AffectedByCastSpeed: true,
+			OnSnapshot: func(sim *core.Simulation, target *core.Unit, dot *core.Dot, isRollover bool) {
+				dot.Snapshot(target, 0.175*Mage.ScalingBaseDamage)
+			},
+			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
+				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeTick)
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			spell.Dot(target).ApplyOrReset(sim)
+			Mage.PyroblastDot.SpellMetrics[target.UnitIndex].Casts = 0
+		},
+	})
+
+	Mage.PyroblastDotImpact = Mage.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 11366}.WithTag(1),
+		SpellSchool:    core.SpellSchoolFire,
+		ProcMask:       core.ProcMaskSpellDamage,
+		Flags:          mage.SpellFlagMage,
+		ClassSpellMask: mage.MageSpellPyroblastDot,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				NonEmpty: true,
+			},
+		},
+
+		DamageMultiplier: 1,
+		CritMultiplier:   Mage.DefaultSpellCritMultiplier(),
+		ThreatMultiplier: 1,
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "PyroblastDoT Fake Impact",
+			},
+			NumberOfTicks:       4,
+			TickLength:          time.Second * 3,
+			AffectedByCastSpeed: true,
+			OnSnapshot: func(sim *core.Simulation, target *core.Unit, dot *core.Dot, _ bool) {
+				//
+			},
+			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
+				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeTick)
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			spell.Dot(target).ApplyOrReset(sim)
+			Mage.PyroblastDot.SpellMetrics[target.UnitIndex].Casts = 0
+		},
+	})
```

Comment:
Why do you need a fake spell?

Reply:
This was a testing thing for the Impact talent. My thought process was to have an Impact version of the spells that could be filled out each time it needed to be spread since it had custom duration. Ended up scrapping that idea half way through and am looking to find a way to duplicate existing auras onto the applicable targets instead, but haven't had time.